command/agent/consul/connect.go

```diff
@@ -200,17 +215,19 @@ func connectNetworkInvariants(networks structs.Networks) error {
 // connectPort returns the network and port for the Connect proxy sidecar
 // defined for this service. An error is returned if the network and port
 // cannot be determined.
-func connectPort(serviceName string, networks structs.Networks) (*structs.NetworkResource, structs.Port, error) {
+func connectPort(portLabel string, networks structs.Networks, ports structs.AllocatedPorts) (structs.AllocatedPortMapping, error) {
 	if err := connectNetworkInvariants(networks); err != nil {
-		return nil, structs.Port{}, err
+		return structs.AllocatedPortMapping{}, err
 	}
-
-	port, ok := networks[0].PortForService(serviceName)
+	mapping, ok := ports.Get(portLabel)
 	if !ok {
-		return nil, structs.Port{}, fmt.Errorf("No Connect port defined for service %q", serviceName)
+		mapping = networks.Port(portLabel)
+		if mapping.Value > 0 {
+			return mapping, nil
+		}
+		return structs.AllocatedPortMapping{}, fmt.Errorf("invalid port %q: port label not found", portLabel)
```

Comment:
To match with similar errors
```suggestion
		return structs.AllocatedPortMapping{}, fmt.Errorf(""No port of label %q defined", portLabel)
```